gst/gst_debug.go

```diff
@@ -201,13 +233,25 @@ type LogFunction = func(
 	message string,
 )
 
+type DebugLogFunction = func(
+	category *DebugCategory,
+	level DebugLevel,
+	file string,
+	function string,
+	line int,
+	object *LoggedObject,
+	message *DebugMessage,
+)
+
 var (
-	logFnMu           sync.RWMutex
-	customLogFunction LogFunction
+	logFnMu                sync.RWMutex
+	customLogFunction      LogFunction
+	customDebugLogFunction DebugLogFunction
 )
 
 // SetLogFunction is used to replace gstreamer's default logging function.
 // Setting to nil will revert to the default logging function.
+// Deprecated: Passed glib.Object to LogFunction can cause broken behavior. Use SetDebugLogFunction instead
```

Comment:
Sorry for the hassle, but I do think we should either fix or break broken code. If your change was just an improvement of the current implementation then breaking it would have been unnecessary.

Since it is fundamentally broken though, we should break the API to "force" the users into fixing it.

Reply:
@RSWilli I'm ok either way

Should we just change the SetLogFunction signature or also remove existing SetLogFunction/LogFunction and leave the new naming convention?

Reply:
I think we've got 10 hour timezone difference between us, but I made an assumption that you're ok changing/breaking the LogFunction signature and squashed the earlier iterations together. But feel free to adjust the code if you want!

Reply:
> 10 hour timezone difference

Yep that is the main reason for the delays, sorry.

But still your assumption was correct!